### PR TITLE
DM-32453: Add over-temperature rule

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: doc/version_history.rst [hunk 1]
```diff
@@ -11,16 +11,20 @@ v1.6.0
 
 Changes:
 
-* Add rule `rules.DewPointDepression`.
-  This rule requires ts_xml 10.1.
-* Add rule `rules.Humidity`.
-* Add rule `rules.MTCCWFollowingRotator`: warn when the MT camera cable wrap is not following the camera rotator.
-* Add classes  `FieldWrapperList`, `BaseFilteredFieldWrapper`, `FilteredFieldWrapper`, and `IndexedFilteredFieldWrapper`.
+* Add rules (most of which require ts_xml 10.0):
```
Comment (wvreeven) [inline doc/version_history.rst:14]: In line 49 you put `ts_xml 10.1`. Which one is it?

File: python/lsst/ts/watcher/filtered_field_wrapper.py [hunk 2]
```diff
@@ -123,49 +119,120 @@ def _get_nelts(self, data):
...
         Watcher model.
     topic : `lsst.ts.salobj.ReadTopic`
         Topic to read.
-    filter_field : `str`
-        Name of filter field.
-    filter_value : `str`
-        Required value of the filter field.
+    sensor_name : `str`
+        Sensor name to filter on.
+        This becomes filer_value in the base class,
```
Comment (wvreeven) [inline python/lsst/ts/watcher/filtered_field_wrapper.py:158]: `filer_value` -> `filter_value`.

File: python/lsst/ts/watcher/rules/dew_point_depression.py [hunk 3]
```diff
@@ -102,30 +117,37 @@ def __init__(self, config):
...
 $schema: http://json-schema.org/draft-07/schema#
-description: Configuration for Enabled
+description: >-
+    Configuration for DewPointDepression rule.
+    A typical closure limit is 2 C. A typical warning level is 3 C.
 type: object
 properties:
   name:
-    description: 'Telescope being monitored, typically AuxTel or MainTel'
+    description: Telescope being monitored, typically AuxTel or MainTel
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/dew_point_depression.py:126]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/dew_point_depression.py [hunk 4]
```diff
@@ -102,30 +117,37 @@ def __init__(self, config):
...
     type: array
     minItems: 1
     items:
       type: object
       properties:
-        sal_name:
-          description: 'SAL component name[:SAL index] e.g. ESS:1'
-          type: string
+        sal_index:
+          description: SAL index of ESS CSC
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/dew_point_depression.py:139]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/dew_point_depression.py [hunk 5]
```diff
@@ -138,34 +160,35 @@ def get_schema(cls):
...
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/dew_point_depression.py:177]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/humidity.py [hunk 6]
```diff
@@ -99,30 +104,32 @@ def __init__(self, config):
...
 $schema: http://json-schema.org/draft-07/schema#
-description: Configuration for Enabled
+description: >-
+    Configuration for Humidity rule.
+    A typical warning level is 73%. It is unusual to have a closing limit.
 type: object
 properties:
   name:
-    description: 'Telescope being monitored, typically AuxTel or MainTel'
+    description: Telescope being monitored, typically AuxTel or MainTel
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/humidity.py:113]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/humidity.py [hunk 7]
```diff
@@ -99,30 +104,32 @@ def __init__(self, config):
...
     type: array
     minItems: 1
     items:
       type: object
       properties:
-        sal_name:
-          description: 'SAL component name[:SAL index] e.g. ESS:1'
-          type: string
+        sal_index:
+          description: SAL index of ESS CSC
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/humidity.py:123]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/over_temperature.py [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+$schema: http://json-schema.org/draft-07/schema#
+description: Configuration for OverTemperature rule.
+type: object
+properties:
+  name:
+    description: System being monitored, e.g. "MT Camera Hexapod"
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/over_temperature.py:114]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/over_temperature.py [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+        ESS temperature sensors to monitor.
+        These can be any topic with a "temperature" field,
+        including temperature, hx85a and hx85ba.
+    type: array
+    minItems: 1
+    items:
+      type: object
+      properties:
+        sal_index:
+          description: SAL index of ESS CSC
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/over_temperature.py:127]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/over_temperature.py [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+                type: string
+              sensor_names:
+                description: List of dict of sensor_name, indices.
+                type: array
+                minItems: 1
+                items:
+                  type: object
+                  properties:
+                    sensor_name:
+                      description: value of sensorName field
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/over_temperature.py:148]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/rules/over_temperature.py [hunk 11]
```diff
@@ -0,0 +1,297 @@
...
+        If omitted the reason will just describe the value and threshold.
+    type: string
+  hysteresis:
+    description: >-
+        The amount by which temperature (C) must decrease below
+        a severity level before alarm severity is decreased.
+    type: number
+    default: 1
+  poll_interval:
+    description: Time delay between polling updates (second)
```
Comment (wvreeven) [inline python/lsst/ts/watcher/rules/over_temperature.py:206]: Please make sure the description ends in a period.

File: python/lsst/ts/watcher/threshold_handler.py [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+
+import functools
+import math
+
+from lsst.ts.idl.enums.Watcher import AlarmSeverity
+
+from .base_rule import NoneNoReason
+
+
+# Levels must be separated by hysteresis * hysteresis_grow_factor,
```
Comment (wvreeven) [inline python/lsst/ts/watcher/threshold_handler.py:32]: `hysteresis_grow_factor` -> `HYSTERSIS_GROW_FACTOR`

**Review by wvreeven — APPROVED** [review]
Looks really good to me. Please consider correcting the typos that I found.

File: doc/version_history.rst (hunk 1 above)
Comment (r-owen) [reply doc/version_history.rst:14]: It's 10.1. Thank you for catching that.

File: python/lsst/ts/watcher/rules/dew_point_depression.py (hunk 5 above)
Comment (r-owen) [reply python/lsst/ts/watcher/rules/dew_point_depression.py]: I have gone through all descriptions for the three rules I worked on in this ticket and made sure they all end with period.

File: python/lsst/ts/watcher/threshold_handler.py (hunk 12 above)
Comment (r-owen) [reply python/lsst/ts/watcher/threshold_handler.py]: I also fixed the spelling of HYSTERESIS (I tend to mis-type and omit the E before the S).

**Pull request merged**